### PR TITLE
Always place vectorized domains innermost

### LINKED ISSUE #2464: Move vectorized domains innermost always
https://github.com/csarofeen/pytorch/blob/7b37a836bc0e55ae17f38d7b92974dd6ef380718/third_party/nvfuser/csrc/scheduler/reduction_utils.cpp#L259-L276

### PR DESCRIPTION
**Pull request by naoyam**
Closes #2464 

All tests are green. 

Checking benchmark performances.

### CODE REVIEW

**Comment by naoyam** [comment]
Didn't expect, but resulted in large-degree performance loss:

image

One extreme case is:

<html>
<body>


NvFuserScheduler_Softmax_Outer_fp16___GRAPH/NvFuserScheduler_Softmax_Outer_fp16/8/33554432/manual_time | 1.323790e+12 | 4.683680e+11
-- | -- | --


<br class="Apple-interchange-newline">
</body>
</html>

It was around 1.3 TB/s and is just 468 GB/s with this PR. It seems it's due to increased register usage as vectorized domains are located inside CA positions. And this particular case resulted in using 68 registers, whereas previously used 58 registers. While the difference is not significant, it triggered to cross over an occupancy cliff, resulting in half of the original occupancy.

It's also interesting that most of the perf loss happen only with softmax benchmarks, so that probably has something to do with the structure of persistent reductions. 

Since this is a non trivial heuristic decision, and it seems mostly negative as is, I'm going to close the PR for now.
